specification/under_development/profiles/Base_Profile.md

```diff
@@ -0,0 +1,175 @@
+
+TODO: why the string value, and not an integer?   
+TODO: no double underscore guard for attributes likely makes sense
+
+### Function Body
+
+The function body consists of a single block named `entry`. This implies in particular that no branching is permitted inside a function's body.
+In the `entry` block, any number of calls to QIS functions may be performed.
+To be compatible with the Base Profile these functions must return void. 
+Any arguments to invoke them must be inlined into the call itself; 
+they hence must be constants or a pointer to a [qubit or result]()
+The result of the program execution should be logged using output recording functions.
+
+The following instructions are the *only* LLVM instructions that are permitted within a Base Profile compliant program:
+
+| LLVM Instruction                  | Context and Purpose         | Rules for Usage |
+|---------------------------|-------------------|-------------|
+| `call`    | Used within a function block to invoke any one of the declared QIS functions and the output recording functions. | May optionally be preceded by a [`tail` marker](https://llvm.org/docs/LangRef.html#call-instruction). |
+| `ret` | Used to return the exit code of the program. | Must occur (only) as the final instruction of the `entry` block. |
+| `inttoptr` | Used to cast an `i64` integer value to either a `%Qubit*` or a `%Result*`. | May be used as part of a function call only. |
+| `getelementptr inbounds` | Used to create an `i8*` to pass a constant string for the purpose of labeling an output value. | May be used as part of call to an output recording function only. |
+
+## Data Types and Values
+
+Within the base profile, defining local variables is not supported. Arguments to calls correspondingly are expected to be constant values. 
+
+Qubits and results are passes as a pointer of type `%Qubit*` and `%Result*` respectively, where the pointer itself rather than the memory location identifies the qubit or result: For the purpose of passing them as arguments in function calls, a 64-bit signed integer value is cast to the appropriate pointer type. The signed integer that is case must be in the interval `[0, nrQubits)` for `%Qubit*` and `[0, nrResults)` for `%Result*`, where `nrQubits` and `nrResults` are the required number of qubits and results defined by entry point attributes.
+
+Only the `%Qubit` and `%Result` data types are required to be supported by all backends. 
+
+Integers and double-precision floating point numbers are available as in full QIR;
+however, computations using these numeric types are not available.
+
+### Qubit and Result Usage
+
+runtime has autonomoie over how to represent qubits. also for base profile, the runtime is the only entity that needs to know how to interprete the opaque pointers (deref or not). that info is captured in the target triple/duo.
```

Comment:
```suggestion
The runtime system has autonomy over how to represent qubits. also for base profile, the runtime is the only entity that needs to know how to interpret the opaque pointers (defer or not). that info is captured in the target triple/duo.
```
This contracts the section  "Data Types and Values" where the interpretation of the opaque pointers is unambiguously stated to not be memory locations.

Reply:
I've fleshed out the explanation of this, and hope it is clear. The runtime still has autonomy over how to represent qubits and results internally - the bitcast of a constant to a pointer is merely a way to reflect application intent.